src/ashley/views.py

```diff
@@ -69,16 +69,25 @@ def _do_on_login(self, lti_request: LTI) -> HttpResponse:
                 type=Forum.FORUM_POST,
                 lti_contexts__id=context.id,
             )
-            # If course is locked, security to remove forum permission on default group
-            # only needed if the course has been marked outside web navigation
-            if context.is_marked_locked and any(
-                item in DEFAULT_FORUM_BASE_READ_PERMISSIONS
+            has_perm = any(
```

Comment:
took me too much time to understand and I think it would have helped if this variable had been called:
```suggestion
            has_any_write_permission = any(
```

Then now that I understand I think this value is correct to check if we are out of sync when the context is marked locked. But what we need to check if the context is marked unlocked is:
```python
 all(item in DEFAULT_FORUM_BASE_WRITE_PERMISSIONS for ...) 
```
right?